File: 075.Sort Colors/075.Sort Colors.cpp
【最直观：平移插入】
class Solution {
public:
    void sortColors(int A[], int n) {
        int i = -1;
        int j = -1;
        int k = -1;
        for(int p = 0; p < n; p ++)
        {
            //根据第i个数字，挪动0~i-1串。
            if(A[p] == 0)
            {
                A[++k] = 2;    //2往后挪
                A[++j] = 1;    //1往后挪
                A[++i] = 0;    //0往后挪
            }
            else if(A[p] == 1)
            {
                A[++k] = 2;
                A[++j] = 1;
            }
            else
                A[++k] = 2;
        }

    }
};

File: 075.Sort Colors/075.Sort Colors.py
#--------------辰星-------------------
class Solution:
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        if not nums: return None
        bucket = {}
        for item in nums:
            bucket[item] = bucket.get(item, 0) + 1
        pos = 0
        for item in sorted(bucket.keys()):         
            limit = pos + bucket[item]            
            nums[pos:limit] = [item] * bucket[item]
            pos = limit